src/Simplic.FileStructure/Interface/IFileStructureDocumentPathRepository.cs

```diff
@@ -15,5 +15,21 @@ public interface IFileStructureDocumentPathRepository : IRepositoryBase<Guid, Fi
         /// <param name="documentId">Document id</param>
         /// <returns>Enumerable of paths</returns>
         IEnumerable<FileStructureDocumenPath> GetByDocumentId(Guid documentId);
+        
+        /// <summary>
+        /// Gets all by Directory id 
+        /// </summary>
+        /// <param name="directoryId"></param>
+        /// <returns></returns>
+        IEnumerable<FileStructureDocumenPath> GetByDirectoryId(Guid directoryId);
+
+        /// <summary>
+        /// Gets a bool if any of the directory id is protected
+        /// It checks all the guids and if any guid is protected it returns true.

```

Comment:
`Goes through a list of directory ids and checks, whether any directory contains a document, that is protected. If a protected documents exists, true will be returned.
If no protected directory was found, false will be returned.`